static/js/components/FollowupRequestForm.jsx

```diff
@@ -0,0 +1,150 @@
+import React, { useState, useEffect } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import PropTypes from "prop-types";
+import Select from "@material-ui/core/Select";
+import InputLabel from "@material-ui/core/InputLabel";
+import MenuItem from "@material-ui/core/MenuItem";
+import Form from "@rjsf/material-ui";
+import * as Actions from "../ducks/source";
+
+import "react-datepicker/dist/react-datepicker-cssmodules.css";
+
+const FollowupRequestForm = ({
+  obj_id,
+  instrumentList,
+  instrumentFormParams,
+}) => {
+  const dispatch = useDispatch();
+  const { telescopeList } = useSelector((state) => state.telescopes);
+  const { allocationList } = useSelector((state) => state.allocations);
+  const allGroups = useSelector((state) => state.groups.all);
+  const [selectedInstrumentId, setSelectedInstrumentId] = useState(null);
+  const [selectedAllocationId, setSelectedAllocationId] = useState(null);
+  const allocationInstIds = allocationList.map((a) => a.instrument_id);
+  const allocatedInstrumentIds = Object.keys(instrumentFormParams)
```

Comment:
So the latter is the list of instrument IDs in the form params that match the list of allocation inst IDs in the DB? Maybe an in-line comment here would help, still feels a bit cryptic

Reply:
added:

```
  // list of the instrument IDs corresponding to each element of the
  // allocationList (all visible allocations to user)
  const allocationInstIds = allocationList.map((a) => a.instrument_id);

  // list of the instrument IDs corresponding to each element of
  // instrumentFormParams (list of instrument APIs).
  const idsOfInstrumentswithAPIs = Object.keys(instrumentFormParams)
    .map((k) => parseInt(k, 10))
    .filter((i) => allocationInstIds.includes(i));
```